task02.rb

```diff
@@ -0,0 +1,28 @@
+#2. For a given array that contains some strings output the content of those strings separated with whitespace between them, having capitalized first word and ending with ".", e.g. [[" thIs ", "Is"], nil, " My  ", ["   first"], "", :program] should output "This is my first program."
+
+def task02(arr)
+  if arr.class != Array
+    raise ArgumentError, "The input type is invalid, expect an array"
+  end
+  if arr.empty?
+    raise ArgumentError, "The input is empty."
+  end
+
+  format_string = arr.flatten.join(" ").strip!
+  if format_string.nil?
+    raise ArgumentError, "Not a valid input. The input had only empty data."
+  end
+  if !format_string.scan(/([^\s\w,-]|_)/).empty?
+    raise ArgumentError, "Not a valid input. Letters, numbers, commas and dash are allowed in logical order."
+  end
+  if !format_string.scan(/\A[,]/).empty?
+    raise ArgumentError, "Not a valid input. A sentence cannot start with a comma. Only letters, numbers and dash are allowed."
+  end
+  if !format_string.scan(/[,-]\z/).empty?
```

Comment:
@MaraEfimova Can't we simply get the last symbol and check it insted of using RegEx?